test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -103,9 +103,11 @@ call_log_manager::call_log_begin_transaction(const json &call_log_entry)
 
     int ret = session->begin_transaction(config);
 
+#ifdef DEBUG
```

Comment:
I assume what's happening here is we'll only populate `call_log_entry` [when `-DHAVE_CALL_LOG=1`](https://github.com/wiredtiger/wiredtiger/blob/0f8e0c26738a0497c7fa353b8de606e1bac23849/src/session/session_api.c#L1828), and `HAVE_CALL_LOG` [can only be true when `HAVE_DIAGNOSTIC=1`](https://github.com/wiredtiger/wiredtiger/blob/0f8e0c26738a0497c7fa353b8de606e1bac23849/cmake/configs/base.cmake#L110). Building in Release mode disables `HAVE_DIAGNOSTIC` and our asserts start failing as there's nothing to fetch from `call_log_entry`. 

There's probably a discussion to be had about the coupling between `CMAKE_BUILD_TYPE=Debug` and `HAVE_DIAGNOSTIC`, but for now I think we can just replace these `#ifdef DEBUG`s with `#ifdef HAVE_CALL_LOG` and things should (should) run fine. 
Pinging one of our original authors @Siddhartha8899 @Sean04 to confirm the above.

Reply:
This is different I think. The Release build will set NDEBUG which turns off assert so:

    int ret_expected = call_log_entry["return"]["return_val"].get<int>();

Is now a variable that is not used and we fail to compile.

Reply:
@ajmorton It is possible to run the simulator in release mode. If the compilation is failing, I do not mind this change.

Reply:
Gotcha, thanks guys.
Another alternative would be replacing the `assert`s with `testutil_assert`, but happy to resolve the convo either way.